### DISCUSSION TITLE
No instructions on mounting a volume on the host

### DISCUSSION THREAD

**Discussion by grkravi** [longhorn/longhorn#7816] [Q&A] [answered]
I am using Longhorn v1.5.3 on k3s v1.26.11+k3s2 with OS - Ubuntu 22.04.3 LTS and also Rancher. I dont see any clear instructions on how to access a longhorn vloume on a node for direct bulk copy of files into longhorn. I can attach a volume to a node but its not showing up on the node itself with lsblk -f or fdisk -l. Please help.

**Comment by c3y1huang** [comment, marked as answer]
If you haven't created the workload attaching the volume, then it's likely the volume is still unformatted.

You can try:
1. Create a bare pod (or workload in other forms) with Longhorn volume, for example: 
    ```bash
    > curl https://raw.githubusercontent.com/longhorn/longhorn/v1.5.3/examples/pod_with_pvc.yaml | kubectl apply -f -
    ```
1. Wait for the pod to run then delete it, this leaves the volume in the detached state.
    ```
    > k get pod
    NAME            READY   STATUS    RESTARTS   AGE
    volume-test     1/1     Running   0          29s
   
    > k delete pod volume-test 
    pod "volume-test" deleted
    
    > k -n longhorn-system get volume
    NAME                                       DATA ENGINE   STATE      ROBUSTNESS   SCHEDULED   SIZE         NODE   AGE
    pvc-9c1d2d7f-28b7-4403-9558-130c56d84878   v1            detached   unknown                  2147483648          68s
    ```
1. Attach the volume to a node via the UI.
1. SSH into the attached node and mount the volume.
    ```
    > mount /dev/longhorn/pvc-9c1d2d7f-28b7-4403-9558-130c56d84878 /mnt/
    ```
1. Write data to the mount point.
    ```bash
    > echo foo > /mnt/bar
    ```
1. Unmount the mount point when done.
    ```bash
    > umount /mnt 
    ```
1. Detach the volume from the UI.
    ```bash
    > k -n longhorn-system get volume
    NAME                                       DATA ENGINE   STATE      ROBUSTNESS   SCHEDULED   SIZE         NODE   AGE
    pvc-9c1d2d7f-28b7-4403-9558-130c56d84878   v1            detached   unknown                  2147483648          7m28s
    ```
1. Recreate the pod (workload).
    ```bash
    > curl https://raw.githubusercontent.com/longhorn/longhorn/v1.5.3/examples/pod_with_pvc.yaml | kubectl apply -f -
    ```
1. Check the data directory in the pod.
    ```bash
    > k exec -it volume-test -- ls /data
    bar         lost+found
    ```

**Reply by grkravi** [reply]
I observed these steps are not working for a RWX volume. I created a new RWX volume from Longhorn UI, and created PV and PVC from UI. Attached to the volume-test pod. After it started running, deleted this pod and attached to a node. When I ssh into that node I am not seeing it in /dev/longhorn. Am I missing something here?

**Reply by derekbit** [reply]
@grkravi You can check the mechanism of RWX volumes in https://longhorn.io/docs/1.5.3/advanced-resources/rwx-workloads/.

**Reply by grkravi** [reply]
Okay, thanks. I am able to copy local files into this RWX volume using "kubectl cp"  command.
